src/scmdata/groupby.py

```diff
@@ -23,59 +35,71 @@ def _maybe_wrap_array(original, new_array):
         return new_array
 
 
-class _GroupBy(ImplementsArrayReduce):
-    def __init__(self, meta, groups, na_fill_value=-10000):
-        m = meta.reset_index(drop=True)
+class RunGroupBy(ImplementsArrayReduce, Generic[GenericRun]):
```

Comment:
I couldn't use the generic with GenericRun inside a `TYPE_CHECKING` block which is bit annoying as that means that RunGroupBy needs to be lazy loaded inside `run.py`

Reply:
That's an option that I overlooked so far for dealing with import cycles. Sounds sensible here.